server.js

```diff
@@ -1,35 +1,308 @@
+      "/tracks/id/:id":"Get a track by _id (for example: /tracks/id/6283e58592aafedccbeb0e0e)",
+      "/tracks?genre=${genre}":"Get tracks by genre (for example: /tracks?genre=hip hop)",
+      "/tracks?artist=${artist}":"Get all tracks from a specific artist (for example: /tracks?artist=billie eilish)",
+      "/tracks?songtitle=${songtitle}":"Get a specific track by the songtitle (for example: /tracks?songtitle=shallow)"
+    }]
+  })
+})
+
+
+// get all tracks
+app.get("/tracks", async (req, res) => {
+  try {
+    const allTracks = await Track.find().sort({id:1})
+    res.status(200).json({
+      data: allTracks,
+      succes: true
+    })
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get tracks by genre OR get tracks by artist name OR get tracks by songtitle
+app.get("/tracks", async (req, res) => {
+  try { 
+    let tracks = await Track.find(req.query)
+
+    if (req.query.genre) {
+      const tracksByGenre = await Track.find({
+        genre: new RegExp(req.query.genre, "i")
+      })
+      tracks = tracksByGenre
+    }
+
+    if (req.query.artist) {
+      const tracksByArtist = await Track.find({
+        artistName: new RegExp(req.query.artist, "i")
+      })
+      tracks = tracksByArtist
+    }
+
+    if (req.query.songtitle) {
+      const tracksBySongtitle = await Track.find({
+        trackName: new RegExp(req.query.songtitle, "i")
+      })
+      tracks = tracksBySongtitle
+    }
+
+    res.status(200).json({
+      data: tracks,
+      success: true
+    })
+    } catch(err) {
+      res.status(400).json({
+        error: "Invalid request",
+        success: false
+      })
+    }
+})
+
+
+// get all tracks with high-tempo, ie. tracks with bpm >= 120
+app.get("/tracks/high-tempo", async (req, res) => {
+  try {
+    const tracksWithHighTempo = await Track.find({
+      bpm: { $gte: 120 }
+    })
+    res.status(200).json({
+      data: tracksWithHighTempo,
+      success: true
+    })
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get all danceable tracks, ie tracks with danceability >= 60
+app.get("/tracks/danceable", async (req, res) => {
+  try {
+    const danceableTracks = await Track.find({
+      danceability: { $gte: 60 }
+    })
+    res.status(200).json({
+      data: danceableTracks,
+      success: true
+    })
+  } catch(err) {
+    res.status(400).json({
+      error: "Invalid request",
+      success: false
+    })
+  }
+})
+
+
+// get tracks by genre
+app.get("/tracks/genres/:genre", async (req, res) => {
+  try {
+    const { genre } = req.params
+    const tracksByGenre = await Track.find({
+      genre: new RegExp(genre, "i")
+    })
+    if (tracksByGenre) {
+      res.status(200).json({
+        data: tracksByGenre,
+        success: true
```

Comment:
I can't make the endpoints with the query work, genre, artist, and songtitle...
I tested them out in your previous week's project and there they worked. 
From the live session on Wednesday Daniel goes through this, maybe take a look at that video and see if you can figure out the problem :)